### ISSUE TITLE
Volume ID is generated on every instance/node

### ISSUE THREAD

**Issue by prashanthpai** [gluster/glusterd2#191]
During volume creation, the volume ID is generated in following call flow:

`validateVolumeCreate() -> createVolinfo() -> volume.NewVolumeEntry()`

The function `validateVolumeCreate()` is run on all instances/nodes. Hence the volinfo set in context (store really) is over-written by each node. Also the `trusted.glusterfs.volume-id` set is different for each brick directory of the volume!

**Comment by prashanthpai** [comment]
~~~sh
[ppai@gd2-1 ~]$ sudo getfattr -d -m . -e hex /export/brick{1..4}/data
getfattr: Removing leading '/' from absolute path names
# file: export/brick1/data
security.selinux=0x756e636f6e66696e65645f753a6f626a6563745f723a756e6c6162656c65645f743a733000
trusted.glusterfs.volume-id=0x9f136993451746568f8f739ad94a0793

# file: export/brick2/data
security.selinux=0x756e636f6e66696e65645f753a6f626a6563745f723a756e6c6162656c65645f743a733000
trusted.glusterfs.volume-id=0x5eb43eb81731455287e98db320e00eb4

# file: export/brick3/data
security.selinux=0x756e636f6e66696e65645f753a6f626a6563745f723a756e6c6162656c65645f743a733000
trusted.glusterfs.volume-id=0x2e625ce668ee43aca7e2eed973e8bac6

# file: export/brick4/data
security.selinux=0x756e636f6e66696e65645f753a6f626a6563745f723a756e6c6162656c65645f743a733000
trusted.glusterfs.volume-id=0x5eb43eb81731455287e98db320e00eb4
~~~

As seen above, every node generates a different volume id. To fix this, only the initiator node should generate the volume id once, set it in transaction context, so that every other node/instance can make use of that.

**Comment by prashanthpai** [comment]
After fix in PR #192 

```sh
[ppai@gd2-1 ~]$ sudo getfattr -d -m . -e hex /export/brick{1..4}/data
getfattr: Removing leading '/' from absolute path names
# file: export/brick1/data
security.selinux=0x756e636f6e66696e65645f753a6f626a6563745f723a756e6c6162656c65645f743a733000
trusted.glusterfs.volume-id=0x599e98f4cced4baa96d67799aa7859dc

# file: export/brick2/data
security.selinux=0x756e636f6e66696e65645f753a6f626a6563745f723a756e6c6162656c65645f743a733000
trusted.glusterfs.volume-id=0x599e98f4cced4baa96d67799aa7859dc

# file: export/brick3/data
security.selinux=0x756e636f6e66696e65645f753a6f626a6563745f723a756e6c6162656c65645f743a733000
trusted.glusterfs.volume-id=0x599e98f4cced4baa96d67799aa7859dc

# file: export/brick4/data
security.selinux=0x756e636f6e66696e65645f753a6f626a6563745f723a756e6c6162656c65645f743a733000
trusted.glusterfs.volume-id=0x599e98f4cced4baa96d67799aa7859dc
```
